src/libraries/System.Net.Quic/tests/FunctionalTests/QuicListenerTests.cs

```diff
@@ -156,13 +158,60 @@ public async Task AcceptConnectionAsync_SlowOptionsCallback_TimesOut(bool useCan
             };
             await using QuicListener listener = await CreateQuicListener(listenerOptions);
 
-            ValueTask<QuicConnection> connectTask = CreateQuicConnection(listener.LocalEndPoint);
-            Exception exception = await AssertThrowsQuicExceptionAsync(QuicError.ConnectionTimeout, async () => await listener.AcceptConnectionAsync());
-            Assert.Equal(SR.Format(SR.net_quic_handshake_timeout, QuicDefaults.HandshakeTimeout), exception.Message);
+            QuicClientConnectionOptions clientOptions = CreateQuicClientOptions(listener.LocalEndPoint);
+            clientOptions.HandshakeTimeout = clientShorterTimeout ? TimeSpan.FromSeconds(2) : TimeSpan.FromSeconds(20);
```

Comment:
I'm pretty sure this will eventually pop-up in CI report. Especially if you're asserting specific types of exception that should happen only in certain order of events depending on this timing.

But we can wait until we see it.